zerver/lib/actions.py

```diff
@@ -3066,6 +3079,16 @@ def get_non_subscribed_tups() -> List[Tuple[UserProfile, Stream]]:
         streams_by_user[sub.user_profile_id].append(stream)
         altered_user_dict[stream.id].append(sub.user_profile)
 
+    for (user_profile, stream) in removed_stream_admin_dict:
+        event = dict(type="subscription",
+                     op="update",
+                     email=user_profile.email,
+                     property="role",
+                     value=Subscription.ROLE_MEMBER,
+                     stream_id=stream.id,
+                     name=stream.name)
+        send_event(user_profile.realm, event, can_access_stream_user_ids(stream))
```

Comment:
@timabbott At this place database queries for `can_access_stream_user_ids` are performed in a loop. I am not able to figure out how to avoid this queries in loop.